ospackage/recipes/buildcommand.go

```diff
@@ -46,10 +49,65 @@ func BuildCommand(step *osconfigpb.SoftwareRecipe_Step, artifacts map[string]str
 
 // StepFileCopy builds the command for a FileCopy step
 func StepFileCopy(step *osconfigpb.SoftwareRecipe_Step_FileCopy, artifacts map[string]string) ([]string, error) {
-	fmt.Println("StepFileCopy")
+	dest, err := normPath(step.FileCopy.Destination)
+	if err != nil {
+		return nil, err
+	}
+
+	permissions, err := parsePermissions(step.FileCopy.Permissions)
+	if err != nil {
+		return nil, err
+	}
+
+	if _, err := os.Stat(dest); err != nil {
+		if !os.IsNotExist(err) {
+			return nil, err
+		}
+	} else {
+		// file exists
+		if !step.FileCopy.Overwrite {
+			logger.Infof("skipping FileCopy step as file at %s already exists", dest)
```

Comment:
I think that this matches the current proto description.

Reply:
i see. i think the proto should be updated, then. it doesn't make sense to succeed in that case, and it's my fault for not catching the comment detail.

to give an example, if a user has a recipe that first copies a file then runs a command which may involve that file, they will have an undefined behavior if we silently pass because the destination exists. if people want to add logic around their copies, they can use script steps.

Reply:
okay. I'll change it to return an error then